fish-rust/src/env.rs

```diff
@@ -35,3 +35,17 @@ pub mod status {
     pub const ENV_INVALID: i32 = 3;
     pub const ENV_NOT_FOUND: i32 = 4;
 }
+
+#[repr(u16)]
+pub enum env_mode_flags_t {
+    ENV_DEFAULT = 0,
+    ENV_LOCAL = 1 << 0,
+    ENV_FUNCTION = 1 << 1,
+    ENV_GLOBAL = 1 << 2,
+    ENV_UNIVERSAL = 1 << 3,
+    ENV_EXPORT = 1 << 4,
+    ENV_UNEXPORT = 1 << 5,
+    ENV_PATHVAR = 1 << 6,
+    ENV_UNPATHVAR = 1 << 7,
+    ENV_USER = 1 << 8,
+}
```

Comment:
Can you please use idiomatic rust naming here? `EnvMode` as the name and `Default`, `Local`, etc. as the values.

Reply:
These constants already exist a few lines above, as `pub const`.
I'm not sure if it should be an enum, does that make it easy to write something like

    has_scope = mode & (ENV_LOCAL | ENV_FUNCTION | ENV_GLOBAL | ENV_UNIVERSAL);

we could also use the bitflags crate.

Reply:
@krobelus  it doesn’t - bitflags probably makes more sense.

Reply:
The reason I defined this enum again is because the above definitions are `c_int` whereas they're more appropriate as `u16` which is the most similar to their C++ definition. I do think the bitflags crate makes more sense for these constants.

Reply:
Out of curiosity, which is more appropriate: [`bitflags`](https://lib.rs/crates/bitflags) or [`enumflags2`](https://lib.rs/crates/enumflags2)? @krobelus @mqudsi